torchtune/utils/_generation.py

```diff
@@ -67,7 +67,7 @@ def generate(
     max_generated_tokens: int,
     temperature: float = 1.0,
     top_k: Optional[int] = None,
-    eos_id: Optional[int] = None,
+    stop_tokens: Optional[List[int]] = None,
```

Comment:
Make it a set for faster lookup

Reply:
oooooooh leetcode coming in clutch

Reply:
Just did the math and O(2) is in fact slower than O(1)